drivers/sx126x/sx126x_netdev.c

```diff
@@ -126,13 +119,11 @@ static int _recv(netdev_t *netdev, void *buf, size_t len, void *info)
 static int _init(netdev_t *netdev)
 {
     sx126x_t *dev = container_of(netdev, sx126x_t, netdev);
-
     if (sx126x_is_stm32wl(dev)) {
 #if IS_USED(MODULE_SX126X_STM32WL)
-        _dev = netdev;
+        event_callback_init(&sx126x_ev_callback, (void*)_sx126x_handler, (void*)netdev);
```

Comment:
the `netdev` implementation already has its mechanism to offload IRQ. Therefore, this could break such mechanism. I propose to leave it as it was before